### PR TITLE
Fix instructions on reproducing ci errors

### PR DESCRIPTION
**Pull request by potiuk**
The old breeze-legacy used to have a possibility of very easy
reproduction of CI failures by executing the right breeze command
that contained the commit hash of the PR being tested. This has
been broken for some time after we migrated to the new breeze,
but finally it was the time when it was needed again.

This PR brings back the capability by:

* addding --image-tag parameters for tests, shell and start-airflow
  commands
* if --image-tag is specified, then rather than building the
  image, it is pulled using the specified hash
* if --image-tag is specified, the local sources are not mounted
  to breeze when started, but the sources already embedded in the
  image are used ("skipped" set for --mount-sources).
* new "removed" command value is added to --mount-sources, it
  causes breeze command to remove the sources from the image (it
  is used when installing airflow during the tests for specified
  version (it's automatically used when --use-airflow-version
  is used).

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
This has been really useful when analysing #24405 - Would be great to merge #24236 and this one to make it faster to investigate other flaky tests.

**Comment by potiuk** [comment]
It's possible now to use new breeze to reproduce failures from CI more easily :)

File: breeze-complete [hunk 1]
```diff
@@ -173,9 +173,6 @@ function breeze_complete::get_known_values_breeze() {
     -g | --github-repository)
         _breeze_known_values="${_breeze_default_github_repository}"
         ;;
-    -s | --github-image-id)
```
Comment (josh-fell) [inline breeze-complete:176]: Should `github-image-id` also be removed/replaced in `_breeze_long_options` as well?

Comment (potiuk) [reply breeze-complete:176]: Naaa. That's legacy code. I would not worry too much - we are going to delete it shortly

Comment (potiuk) [reply breeze-complete:176]: I hope :dagger:

**Review by josh-fell — APPROVED** [review]

**Pull request merged**